worlds/osrs/__init__.py

```diff
@@ -0,0 +1,616 @@
+                    if "*" not in item_name:
+                        entrance.access_rule = lambda state, item_name=item_name: state.has(item_name, self.player)
+                    else:
+                        self.generate_special_rules_for(entrance, region_row, outbound_region_name)
+                else:
+                    self.generate_special_rules_for(entrance, region_row, outbound_region_name)
+                entrance.connect(self.region_name_to_data[outbound_region_name.replace('*', '')])
+            for resource_region in region_row.resources:
+                if resource_region != "":
+                    entrance = region.create_exit(f"{region_row.name}->{resource_region.replace('*', '')}")
+                    if "*" not in resource_region:
+                        entrance.connect(self.region_name_to_data[resource_region])
+                    else:
+                        self.generate_special_rules_for(entrance, region_row, resource_region)
+                        entrance.connect(self.region_name_to_data[resource_region.replace('*', '')])
+
+        self.roll_locations()
+
+    def generate_special_rules_for(self, entrance, region_row, outbound_region_name):
+        # print(f"Special rules required to access region {outbound_region_name} from {region_row.name}")
+        if outbound_region_name == "Cook's Guild":
+            item_name = self.region_rows_by_name[outbound_region_name].itemReq.replace('*', '')
+            entrance.access_rule = lambda state: state.has(item_name, self.player) and\
+                self.can_reach_skill(state, "cooking", 32)
+            return
+        if outbound_region_name == "Crafting Guild":
+            item_name = self.region_rows_by_name[outbound_region_name].itemReq.replace('*', '')
+            entrance.access_rule = lambda state:  state.has(item_name, self.player) and\
+                self.can_reach_skill(state, "crafting", 40)
+            return
+        if outbound_region_name == "Corsair Cove":
+            item_name = self.region_rows_by_name[outbound_region_name].itemReq.replace('*', '')
+            # Need to be able to start Corsair Curse in addition to having the item
+            entrance.access_rule = lambda state: state.has(item_name, self.player) and \
+                                                 state.can_reach(RegionNames.Falador_Farm, None, self.player)
+            return
+        if outbound_region_name == "Camdozaal*":
+            item_name = self.region_rows_by_name[outbound_region_name.replace('*', '')].itemReq
+            entrance.access_rule = lambda state:  state.has(item_name, self.player) and\
+                state.has(ItemNames.QP_Below_Ice_Mountain, self.player)
+            return
+        if region_row.name == "Dwarven Mountain Pass" and outbound_region_name == "Anvil*":
+            entrance.access_rule = lambda state: state.has(ItemNames.QP_Dorics_Quest, self.player)
+            return
```

Comment:
Oh huh, are the \* on *items* never actually used? Not a big deal, just sth I noticed

Reply:
Aight